src/pkg/toolbox/image.go

```diff
@@ -0,0 +1,131 @@
+package toolbox
+
+import (
+	"errors"
+	"path/filepath"
+
+	"github.com/containers/toolbox/pkg/utils"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	// knownImages holds all known toolbox images
+	// These should work out of the box with all Toolbox commands
+	knownImages = map[string]Image{
+		"fedora-toolbox": {
+			"registry.fedoraproject.org",
+			"fedora-toolbox",
+			"latest",
+		},
+		"ubi7": {
+			"registry.access.redhat.com",
+			"ubi7",
+			"latest",
+		},
+		"ubi7-2": {
+			"registry.access.redhat.com",
+			"ubi7/ubi",
+			"latest",
+		},
+		"ubi8": {
+			"registry.access.redhat.com",
+			"ubi8",
+			"latest",
+		},
+		"ubi8-2": {
+			"registry.access.redhat.com",
+			"ubi8/ubi",
+			"latest",
+		},
+	}
+
+	// supportedSystems holds ids of known systems. IDs are taken from the ID
+	// entry of 'os-release'. Every id has a toolbox image attached.
+	supportedSystems = map[string]Image{
+		"fedora": knownImages["fedora-toolbox"],
+		"rhel":   knownImages["ubi8"],
+	}
+)
+
+// GetFallbackImage returns a Fedora toolbox image
+func GetFallbackImage() Image {
+	return supportedSystems["fedora"]
+}
+
+// GetImageForSystem returns an image matching the given system ID.
+//
+// System ID should be taken from the ID entry of 'os-release'.
+//
+// https://www.freedesktop.org/software/systemd/man/os-release.html#ID=
+func GetImageForSystem(systemID string) (Image, error) {
+	var img Image
+
+	if !IsSystemSupported(systemID) {
+		return img, errors.New("unsupported system")
+	}
+
+	img = supportedSystems[systemID]
+
+	return img, nil
+}
+
+// Image holds parts of a full URI of an image
+type Image struct {
```

Comment:
I am wondering if we should call it `Distro`, instead of `Image`. Then we could include a function pointer to parse the release string in a distro-specific way.